hubverse_annotator/app.py

```diff
@@ -86,79 +86,175 @@ def forecast_annotation_ui(
 
 
 def model_and_target_selection_ui(
-    single_loc_hub_table: pl.DataFrame,
+    observed_data_table: pl.DataFrame,
+    forecast_table: pl.DataFrame,
+    loc_abbr: str,
 ) -> tuple[list[str], str]:
     """
     Streamlit widget for model and target selection.
 
     Parameters
     ----------
-    single_loc_hub_table : pl.DataFrame
+    observed_data_table : pl.DataFrame
+        A hubverse table of loaded data (possibly empty).
+    forecast_table : pl.DataFrame
         The hubverse formatted table of forecasted ED
-        visits and or hospital admissions, filtered by
-        location.
+        visits and or hospital admissions (possibly empty).
+    loc_abbr : str
+        The selection location, typically a US jurisdiction.
 
     Returns
     -------
     tuple
         Returns a list of selected model names and the
         selected target.
     """
-    models = single_loc_hub_table["model"].unique().sort().to_list()
-    selected_models = st.multiselect(
-        "Model(s)", options=models, default=models, key="model_selection"
-    )
-    targets = (
-        single_loc_hub_table.filter(pl.col("model").is_in(selected_models))
-        .get_column("target")
-        .unique()
-        .sort()
-        .to_list()
-    )
+    if not forecast_table.is_empty():
```

Comment:
What is the mechanism by which these tables could be empty? Even if they are empty, can't you do still do the filtering?